src/pybind/mgr/dashboard/frontend/src/app/shared/services/doc.service.ts

```diff
@@ -24,7 +24,7 @@ export class DocService {
   }
 
   urlGenerator(section: string, release = 'master'): string {
-    const domain = `http://docs.ceph.com/docs/${release}/`;
+    const domain = `http://docs.ceph.com/en/${release}/`;
```

Comment:
please see https://tracker.ceph.com/issues/48012#note-6 . so this change is not a must have anymore with the new redirect rules. and i am fine if we don't want this change. as both the "page not found" links are not reachable. but, IMHO, it might be a good idea to encode the locale in the path.

Reply:
In order to return the` latest` version for master, I also change `http` to `https`:

```suggestion
    const docVersion = release === 'master' ? 'latest': release;
    const domain = `https://docs.ceph.com/en/${docVersion}/`;
```

The patch contains the above suggestion and fix to the frontend unit test failure (I also add a new test).
https://gist.github.com/bk201/5a69890b59cdccf1c26169bd2736511a

Reply:
@bk201 thanks Kiefer, i incorporated your change into this change, added your Signed-off-by line and repushed.